replication.drush.inc

```diff
@@ -0,0 +1,272 @@
+  // Array of "Callback arguments" and "command line args".
+  $params = func_get_args();
+  if (isset($params[0]) || isset($params[1])) {
+    replication_command_validate($params);
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_replication_start($source, $target) {
+  try {
+    $client = replication_client_factory();
+    $continuous = drush_get_option('continuous');
+    return $client->replicate($source, $target, NULL, $continuous);
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_replication_stop($source, $target) {
+  try {
+    $client = replication_client_factory();
+    $continuous = drush_get_option('continuous');
+    return $client->replicate($source, $target, TRUE, $continuous);
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ *
+ * Prints information about the specific active replication between target and
+ * source databases.
+ */
+function drush_replication_active($source = NULL, $target = NULL) {
+  try {
+    $client = replication_client_factory();
+    $results = $client->getActiveTasks();
+    foreach ($results as $key => $result) {
+      $results[$key]['started_on'] = date('D, j M Y, H:i:s e', $result['started_on']);
+      if ($source && $target && is_array($results)) {
+        $source_diff = array_diff(replication_get_url_parts($result['source']), replication_get_url_parts($source));
+        $target_diff = array_diff(replication_get_url_parts($result['target']), replication_get_url_parts($target));
+        if (empty($source_diff) && empty($target_diff)) {
+          // Return information about one active replication.
+          return array($results[$key]);
+        }
+        else {
+          drupal_set_message('No active replication.', 'status');
+          return;
+        }
+      }
+    }
+    if (!empty($results)) {
+      // Return information about all active replications.
+      return $results;
+    }
+    else {
+      drupal_set_message('No active replications.', 'status');
+    }
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Helper function for command validation.
+ *
+ * @param $params
+ */
+function replication_command_validate($params) {
+  $replicator = drush_get_option('replicator');
+  if ($replicator && replication_get_http_response_code($replicator) != 200) {
+    drush_set_error(dt('Replicator not found.'));
+  }
+  if (replication_get_http_response_code($params['0']) != 200) {
+    drush_set_error(dt('Source database not found.'));
+  }
+  if (replication_get_http_response_code($params['1']) != 200) {
+    drush_set_error(dt('Target database not found.'));
+  }
+}
+
+/**
+ * Returns the CouchDBClient() object.
+ */
+function replication_client_factory() {
+  $options = array('dbname' => '');
+  $replicator = drush_get_option('replicator');
```

Comment:
We should use a default value for the replicator. So if the user doesn't specify `--replicator=XX` the default value will be to use `http://localhost:5984`.